drivers/flash/flash_stm32_qspi.c

```diff
@@ -875,34 +992,63 @@ static int spi_nor_process_bfp(const struct device *dev,
 			}
 		}
 	}
+
 	/*
-	 * Only check if the 1-4-4 (i.e. 4READ) fast read operation is
-	 * supported - other modes - e.g. 1-1-4 (QREAD) or 1-1-2 (DREAD) are
-	 * not.
+	 * Only check if the 1-4-4 (i.e. 4READ) or 1-1-4 (QREAD)
+	 * is supported - other modes are not.
 	 */
 	if (IS_ENABLED(STM32_QSPI_USE_QUAD_IO)) {
+		const enum jesd216_mode_type supported_modes[] = { JESD216_MODE_114,
+								   JESD216_MODE_144 };
+		struct jesd216_bfp_dw15 dw15;
 		struct jesd216_instr res;
 
-		rc = jesd216_bfp_read_support(php, bfp, JESD216_MODE_144, &res);
-		if (rc > 0) {
-			/* Program flash memory to use SIO[0123] */
-			rc = qspi_program_quad_io(dev);
-			if (rc) {
-				LOG_ERR("Unable to enable QUAD IO mode: %d\n",
-					rc);
-				return rc;
+		/* reset active mode */
+		data->mode = STM32_QSPI_UNKNOWN_MODE;
+
+		/* query supported read modes, begin from the slowest */
+		for (size_t i = 0; i < ARRAY_SIZE(supported_modes); ++i) {
+			rc = jesd216_bfp_read_support(php, bfp, supported_modes[i], &res);
+			if (rc >= 0) {
+				LOG_INF("Quad read mode %d instr [0x%x] supported",
+					supported_modes[i], res.instr);
+				data->mode = supported_modes[i];
+				data->qspi_read_cmd = res.instr;
+				data->qspi_read_cmd_latency = res.wait_states;
+				if (res.mode_clocks) {
+					data->qspi_read_cmd_latency += res.mode_clocks;
+				}
 			}
+		}
 
-			LOG_INF("Mode: 1-4-4 with instr:[0x%x] supported!",
-				res.instr);
+		/* don't continue when there is no supported mode */
+		if (data->mode == STM32_QSPI_UNKNOWN_MODE) {
+			LOG_ERR("No supported flash read mode found");
+			return -ENOTSUP;
+		}
 
-			data->qspi_read_cmd = res.instr;
-			data->qspi_read_cmd_latency = res.wait_states;
-			if (res.mode_clocks) {
-				data->qspi_read_cmd_latency +=
-					res.mode_clocks;
-			}
+		LOG_INF("Quad read mode %d instr [0x%x] will be used", data->mode, res.instr);
+
+		/* decode QE requirement type */
+		/* bypass DTS QE requirement (if possible) */
+		rc = jesd216_bfp_decode_dw15(php, bfp, &dw15);
+		if (rc < 0) {
+			LOG_ERR("Unable to decode QE requirement [DW15]: %d", rc);
```

Comment:
Are we able to continue in this case ?

Reply:
Yes, as `qer_type` can be defined in the DTS - `quad_enable_requirements` and if not, then `JESD216_DW15_QER_VAL_S1B6` is used to stay functionally equal to the driver that had no `quad_enable_requirements` support (_it was hardcoded to `JESD216_DW15_QER_VAL_S1B6`_).